{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# This is a Notebook to run RNAseq analysis server\n",
    "# For FFPE or frozen samples \n",
    "\n",
    "\n",
    "This is a warp up of the few steps for the pre-procesing in the RNAseq analaysis, the notebook include:\n",
    "- fastqc\n",
    "- trimming, trim-galore\n",
    "- Alignment, STAR\n",
    "- Annotation counts, featureCounts\n",
    "- multiqc, QC of the process\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/DATA/home/l.gonzalez/Notebooks\r\n"
     ]
    }
   ],
   "source": [
    "#this is in case you forget where you are and what is in it\n",
    "!pwd\n",
    "#!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os, sys\n",
    "import glob\n",
    "import logging"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Before starting the analysis we will check the QC of the fastq files usign fastqc\n",
    "To do that, please go to the cell below and change the path to the folder where the fastq files are, then run it"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!fastqc /DATA/share/pipelines/lgleon/test/testCode_steps_Nousefuldata/testdata/fastq/*fastq.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run trim_galore for the trimming part paired-end\n",
    "Then remove the adaptors using trim galore (conda)\n",
    "The CLI can be run directly "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#!conda init bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Trimming step go where the fastq raw files are\n",
    "\n",
    "#Binary = \"/home/l.gonzalez/programs/TrimGalore-0.6.5/trim_galore\"\n",
    "#Be aware that the path to the raw files (prj) is hard coded\n",
    "# Change it  !!!!!!!!!!!!\n",
    "#prj is hardcode, need to be change with any new analysis\n",
    "\n",
    "#Script, DO NOT RUN !!!!!!!!!!!!!!!!!\n",
    "#!bin/bash\n",
    "\n",
    "#############################################################################\n",
    "##############################################################################\n",
    "# SINGLE-END part\n",
    "##############################################################################\n",
    "##############################################################################\n",
    "\n",
    "##careful with fastq.gz OR fq.gz   can give problems!\n",
    "\n",
    "##Hardcode fastq folder change when need it\n",
    "#    prj=/DATA/share/pipelines/lgleon/C5_RNAseq_FFPE/fastq_files/\n",
    "\n",
    "#for i in ${prj}/*R1*.f*q.gz \n",
    "\n",
    "#do \n",
    "#    name=$(basename ${i} _R1_001.fastq.gz)\n",
    "\n",
    "#\techo ${name}\n",
    "#\techo ${name}_R1_001.fastq.gz\n",
    "#\techo ${name}_R2_001.fastq.gz\n",
    "#\techo ${prj}trimmed/\n",
    "\n",
    "\n",
    "#    /home/l.gonzalez/programs/TrimGalore-0.6.5/trim_galore -q 35 -o ${prj}trimmed/ ${name}_R1_001.fastq.gz\n",
    "    \n",
    "\n",
    "#done\n",
    "\n",
    "#############################################################################\n",
    "##############################################################################\n",
    "# PAIRED-END part\n",
    "##############################################################################\n",
    "##############################################################################\n",
    "#to work with this part remove the # in the single ones, doubles are explanations\n",
    "\n",
    "##careful with fastq.gz OR fq.gz   can give problems!\n",
    "\n",
    "##Hardcode fastq folder change when need it\n",
    "        prj=/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals\n",
    "\n",
    "for i in ${prj}/*R1*.f*q \n",
    "\n",
    "do \n",
    "    #name=$(basename ${i} _R1_001.fastq.gz)\n",
    "    name=$(basename ${i/%_R1_001.f*q})\n",
    "\n",
    "\techo ${name}\n",
    "    #echo ${name/%_R1_001.f*q}       #filenamestem=${file/%.f*q.gz/}\n",
    "\techo ${name}_R1_001.fastq  #option namefq1=${name}_R1_001.fastq.gz, then in the cli use ${namefq1}\n",
    "\techo ${name}_R2_001.fastq  #option namefq2=${name}_R2_001.fastq.gz\n",
    "    echo ${prj}/${name}_R1_001.fastq\n",
    "    echo ${prj}/trimmNormals/          #option pathoutput=${prj}/trimmed/\n",
    "\n",
    "\n",
    "\n",
    "    /home/l.gonzalez/programs/TrimGalore-0.6.5/trim_galore -q 35 --paired -o ${prj}/trimmNormals/ ${prj}/${name}_R1_001.fastq ${prj}/${name}_R2_001.fastq #> logtrim.txt &\n",
    "    \n",
    "    \n",
    "    #### If you run this from Notebook, to the other folder then ${name} is not enough, that is only the name os the files\n",
    "    ####you need to put ${prj}/${name}_R1_001.fastq.gz to work ??\n",
    "    \n",
    "\n",
    "\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CC15N_CAGATC_L007\n",
      "CC15N_CAGATC_L007_R1_001.fastq\n",
      "CC15N_CAGATC_L007_R2_001.fastq\n",
      "/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC15N_CAGATC_L007_R1_001.fastq\n",
      "/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/trimmNormals/\n",
      "Multicore support not enabled. Proceeding with single-core trimming.\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 3.4\n",
      "single-core operation.\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/trimmNormals/\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC15N_CAGATC_L007_R1_001.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t102403\tAGATCGGAAGAGC\t1000000\t10.24\n",
      "Nextera\t34\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t8\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 102403). Second best hit was Nextera (count: 34)\n",
      "\n",
      "Writing report to '/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/trimmNormals/CC15N_CAGATC_L007_R1_001.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC15N_CAGATC_L007_R1_001.fastq\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.6.4_dev\n",
      "Cutadapt version: 3.4\n",
      "Number of cores used for trimming: 1\n",
      "Quality Phred score cutoff: 35\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 3.4). Setting -j 1\n",
      "Writing final adapter and quality trimmed output to CC15N_CAGATC_L007_R1_001_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 35') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC15N_CAGATC_L007_R1_001.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "This is cutadapt 3.4 with Python 3.8.5\n",
      "Command line parameters: -j 1 -e 0.1 -q 35 -O 1 -a AGATCGGAAGAGC /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC15N_CAGATC_L007_R1_001.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 338.99 s (8 Âµs/read; 7.38 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              41,714,643\n",
      "Reads with adapters:                14,577,907 (34.9%)\n",
      "Reads written (passing filters):    41,714,643 (100.0%)\n",
      "\n",
      "Total basepairs processed: 5,256,045,018 bp\n",
      "Quality-trimmed:           1,344,605,927 bp (25.6%)\n",
      "Total written (filtered):  3,804,862,636 bp (72.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 14577907 times\n",
      "\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.7%\n",
      "  C: 29.7%\n",
      "  G: 25.4%\n",
      "  T: 17.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t6665667\t10428660.8\t0\t6665667\n",
      "2\t2341388\t2607165.2\t0\t2341388\n",
      "3\t679999\t651791.3\t0\t679999\n",
      "4\t312790\t162947.8\t0\t312790\n",
      "5\t198896\t40737.0\t0\t198896\n",
      "6\t240818\t10184.2\t0\t240818\n",
      "7\t231241\t2546.1\t0\t231241\n",
      "8\t167305\t636.5\t0\t167305\n",
      "9\t215412\t159.1\t0\t214983 429\n",
      "10\t137318\t39.8\t1\t135911 1407\n",
      "11\t130973\t9.9\t1\t129479 1494\n",
      "12\t166385\t2.5\t1\t164630 1755\n",
      "13\t150746\t0.6\t1\t148918 1828\n",
      "14\t166520\t0.6\t1\t164244 2276\n",
      "15\t145140\t0.6\t1\t142941 2199\n",
      "16\t148321\t0.6\t1\t145959 2362\n",
      "17\t182608\t0.6\t1\t180010 2598\n",
      "18\t123336\t0.6\t1\t121697 1639\n",
      "19\t127970\t0.6\t1\t126151 1819\n",
      "20\t114520\t0.6\t1\t112813 1707\n",
      "21\t156562\t0.6\t1\t154199 2363\n",
      "22\t115566\t0.6\t1\t113835 1731\n",
      "23\t115088\t0.6\t1\t113312 1776\n",
      "24\t126789\t0.6\t1\t124721 2068\n",
      "25\t107910\t0.6\t1\t106347 1563\n",
      "26\t82573\t0.6\t1\t81338 1235\n",
      "27\t101786\t0.6\t1\t100177 1609\n",
      "28\t106395\t0.6\t1\t104706 1689\n",
      "29\t83445\t0.6\t1\t82152 1293\n",
      "30\t82253\t0.6\t1\t81061 1192\n",
      "31\t68822\t0.6\t1\t67779 1043\n",
      "32\t77576\t0.6\t1\t76430 1146\n",
      "33\t49794\t0.6\t1\t48844 950\n",
      "34\t71962\t0.6\t1\t70776 1186\n",
      "35\t67889\t0.6\t1\t66794 1095\n",
      "36\t58054\t0.6\t1\t57170 884\n",
      "37\t48649\t0.6\t1\t47948 701\n",
      "38\t48742\t0.6\t1\t47993 749\n",
      "39\t33939\t0.6\t1\t33384 555\n",
      "40\t37941\t0.6\t1\t37007 934\n",
      "41\t88665\t0.6\t1\t87961 704\n",
      "42\t5694\t0.6\t1\t5584 110\n",
      "43\t3180\t0.6\t1\t3100 80\n",
      "44\t5202\t0.6\t1\t5122 80\n",
      "45\t13225\t0.6\t1\t13079 146\n",
      "46\t13791\t0.6\t1\t13602 189\n",
      "47\t17188\t0.6\t1\t16972 216\n",
      "48\t10647\t0.6\t1\t10481 166\n",
      "49\t15175\t0.6\t1\t14962 213\n",
      "50\t10529\t0.6\t1\t10355 174\n",
      "51\t13040\t0.6\t1\t12854 186\n",
      "52\t13130\t0.6\t1\t12926 204\n",
      "53\t10634\t0.6\t1\t10522 112\n",
      "54\t3407\t0.6\t1\t3331 76\n",
      "55\t5481\t0.6\t1\t5394 87\n",
      "56\t5889\t0.6\t1\t5781 108\n",
      "57\t5408\t0.6\t1\t5354 54\n",
      "58\t3734\t0.6\t1\t3672 62\n",
      "59\t4156\t0.6\t1\t4084 72\n",
      "60\t3257\t0.6\t1\t3132 125\n",
      "61\t4769\t0.6\t1\t4668 101\n",
      "62\t4125\t0.6\t1\t4053 72\n",
      "63\t3023\t0.6\t1\t2923 100\n",
      "64\t2888\t0.6\t1\t2769 119\n",
      "65\t2389\t0.6\t1\t2296 93\n",
      "66\t2504\t0.6\t1\t2411 93\n",
      "67\t2593\t0.6\t1\t2501 92\n",
      "68\t2374\t0.6\t1\t2242 132\n",
      "69\t2330\t0.6\t1\t2183 147\n",
      "70\t1899\t0.6\t1\t1748 151\n",
      "71\t1545\t0.6\t1\t1391 154\n",
      "72\t1524\t0.6\t1\t1331 193\n",
      "73\t1920\t0.6\t1\t1230 690\n",
      "74\t2339\t0.6\t1\t2125 214\n",
      "75\t468\t0.6\t1\t430 38\n",
      "76\t32\t0.6\t1\t22 10\n",
      "77\t20\t0.6\t1\t4 16\n",
      "78\t15\t0.6\t1\t1 14\n",
      "79\t15\t0.6\t1\t1 14\n",
      "80\t11\t0.6\t1\t1 10\n",
      "81\t16\t0.6\t1\t1 15\n",
      "82\t16\t0.6\t1\t1 15\n",
      "83\t22\t0.6\t1\t0 22\n",
      "84\t9\t0.6\t1\t0 9\n",
      "85\t23\t0.6\t1\t1 22\n",
      "86\t17\t0.6\t1\t0 17\n",
      "87\t11\t0.6\t1\t1 10\n",
      "88\t13\t0.6\t1\t0 13\n",
      "89\t5\t0.6\t1\t1 4\n",
      "90\t6\t0.6\t1\t0 6\n",
      "91\t6\t0.6\t1\t0 6\n",
      "92\t9\t0.6\t1\t0 9\n",
      "93\t14\t0.6\t1\t0 14\n",
      "94\t10\t0.6\t1\t0 10\n",
      "95\t10\t0.6\t1\t0 10\n",
      "96\t10\t0.6\t1\t0 10\n",
      "97\t25\t0.6\t1\t0 25\n",
      "98\t13\t0.6\t1\t0 13\n",
      "99\t15\t0.6\t1\t0 15\n",
      "100\t22\t0.6\t1\t0 22\n",
      "101\t9\t0.6\t1\t0 9\n",
      "102\t5\t0.6\t1\t0 5\n",
      "103\t18\t0.6\t1\t0 18\n",
      "104\t11\t0.6\t1\t0 11\n",
      "105\t12\t0.6\t1\t0 12\n",
      "106\t7\t0.6\t1\t0 7\n",
      "107\t19\t0.6\t1\t0 19\n",
      "108\t30\t0.6\t1\t0 30\n",
      "109\t22\t0.6\t1\t0 22\n",
      "110\t23\t0.6\t1\t0 23\n",
      "111\t14\t0.6\t1\t0 14\n",
      "112\t12\t0.6\t1\t0 12\n",
      "113\t12\t0.6\t1\t0 12\n",
      "114\t27\t0.6\t1\t0 27\n",
      "115\t9\t0.6\t1\t0 9\n",
      "116\t13\t0.6\t1\t0 13\n",
      "117\t15\t0.6\t1\t0 15\n",
      "118\t10\t0.6\t1\t0 10\n",
      "119\t10\t0.6\t1\t0 10\n",
      "120\t9\t0.6\t1\t0 9\n",
      "121\t5\t0.6\t1\t0 5\n",
      "122\t14\t0.6\t1\t0 14\n",
      "123\t11\t0.6\t1\t0 11\n",
      "124\t8\t0.6\t1\t0 8\n",
      "125\t2\t0.6\t1\t0 2\n",
      "126\t5\t0.6\t1\t0 5\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC15N_CAGATC_L007_R1_001.fastq\n",
      "=============================================\n",
      "41714643 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to '/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/trimmNormals/CC15N_CAGATC_L007_R2_001.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC15N_CAGATC_L007_R2_001.fastq\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.6.4_dev\n",
      "Cutadapt version: 3.4\n",
      "Number of cores used for trimming: 1\n",
      "Quality Phred score cutoff: 35\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 3.4). Setting -j -j 1\n",
      "Writing final adapter and quality trimmed output to CC15N_CAGATC_L007_R2_001_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 35') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC15N_CAGATC_L007_R2_001.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "This is cutadapt 3.4 with Python 3.8.5\n",
      "Command line parameters: -j 1 -e 0.1 -q 35 -O 1 -a AGATCGGAAGAGC /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC15N_CAGATC_L007_R2_001.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 330.09 s (8 Âµs/read; 7.58 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              41,714,643\n",
      "Reads with adapters:                13,713,211 (32.9%)\n",
      "Reads written (passing filters):    41,714,643 (100.0%)\n",
      "\n",
      "Total basepairs processed: 5,256,045,018 bp\n",
      "Quality-trimmed:           1,524,635,140 bp (29.0%)\n",
      "Total written (filtered):  3,639,466,626 bp (69.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 13713211 times\n",
      "\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 29.1%\n",
      "  C: 30.8%\n",
      "  G: 25.6%\n",
      "  T: 14.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t6642236\t10428660.8\t0\t6642236\n",
      "2\t1967698\t2607165.2\t0\t1967698\n",
      "3\t719839\t651791.3\t0\t719839\n",
      "4\t289362\t162947.8\t0\t289362\n",
      "5\t202145\t40737.0\t0\t202145\n",
      "6\t239281\t10184.2\t0\t239281\n",
      "7\t236788\t2546.1\t0\t236788\n",
      "8\t169917\t636.5\t0\t169917\n",
      "9\t223130\t159.1\t0\t222492 638\n",
      "10\t159114\t39.8\t1\t157619 1495\n",
      "11\t118047\t9.9\t1\t116619 1428\n",
      "12\t218160\t2.5\t1\t216425 1735\n",
      "13\t119333\t0.6\t1\t118065 1268\n",
      "14\t309766\t0.6\t1\t307942 1824\n",
      "15\t54954\t0.6\t1\t54261 693\n",
      "16\t89068\t0.6\t1\t88161 907\n",
      "17\t213024\t0.6\t1\t211689 1335\n",
      "18\t30376\t0.6\t1\t29877 499\n",
      "19\t112455\t0.6\t1\t111527 928\n",
      "20\t65248\t0.6\t1\t64545 703\n",
      "21\t48426\t0.6\t1\t47827 599\n",
      "22\t113452\t0.6\t1\t112043 1409\n",
      "23\t128855\t0.6\t1\t126849 2006\n",
      "24\t149593\t0.6\t1\t147669 1924\n",
      "25\t46278\t0.6\t1\t45611 667\n",
      "26\t82061\t0.6\t1\t80912 1149\n",
      "27\t72980\t0.6\t1\t72034 946\n",
      "28\t79871\t0.6\t1\t78884 987\n",
      "29\t33631\t0.6\t1\t32998 633\n",
      "30\t102233\t0.6\t1\t101065 1168\n",
      "31\t23197\t0.6\t1\t22778 419\n",
      "32\t51585\t0.6\t1\t50824 761\n",
      "33\t44923\t0.6\t1\t44165 758\n",
      "34\t58024\t0.6\t1\t57224 800\n",
      "35\t40669\t0.6\t1\t40107 562\n",
      "36\t32362\t0.6\t1\t31912 450\n",
      "37\t54904\t0.6\t1\t54171 733\n",
      "38\t26554\t0.6\t1\t26119 435\n",
      "39\t36856\t0.6\t1\t36430 426\n",
      "40\t25272\t0.6\t1\t24931 341\n",
      "41\t35026\t0.6\t1\t34480 546\n",
      "42\t39080\t0.6\t1\t38598 482\n",
      "43\t11770\t0.6\t1\t11559 211\n",
      "44\t23050\t0.6\t1\t22725 325\n",
      "45\t28026\t0.6\t1\t27691 335\n",
      "46\t11742\t0.6\t1\t11576 166\n",
      "47\t8912\t0.6\t1\t8757 155\n",
      "48\t11851\t0.6\t1\t11706 145\n",
      "49\t8844\t0.6\t1\t8683 161\n",
      "50\t9589\t0.6\t1\t9413 176\n",
      "51\t14587\t0.6\t1\t14419 168\n",
      "52\t4329\t0.6\t1\t4240 89\n",
      "53\t4139\t0.6\t1\t4071 68\n",
      "54\t5030\t0.6\t1\t4955 75\n",
      "55\t6429\t0.6\t1\t6326 103\n",
      "56\t4203\t0.6\t1\t4126 77\n",
      "57\t5547\t0.6\t1\t5440 107\n",
      "58\t5373\t0.6\t1\t5283 90\n",
      "59\t4293\t0.6\t1\t4212 81\n",
      "60\t4899\t0.6\t1\t4783 116\n",
      "61\t4686\t0.6\t1\t4574 112\n",
      "62\t4424\t0.6\t1\t4300 124\n",
      "63\t4236\t0.6\t1\t4051 185\n",
      "64\t3532\t0.6\t1\t3383 149\n",
      "65\t3172\t0.6\t1\t3048 124\n",
      "66\t3040\t0.6\t1\t2854 186\n",
      "67\t3545\t0.6\t1\t2738 807\n",
      "68\t6859\t0.6\t1\t6704 155\n",
      "69\t131\t0.6\t1\t107 24\n",
      "70\t14\t0.6\t1\t0 14\n",
      "71\t18\t0.6\t1\t0 18\n",
      "72\t22\t0.6\t1\t1 21\n",
      "73\t22\t0.6\t1\t1 21\n",
      "74\t34\t0.6\t1\t1 33\n",
      "75\t22\t0.6\t1\t1 21\n",
      "76\t21\t0.6\t1\t2 19\n",
      "77\t11\t0.6\t1\t1 10\n",
      "78\t25\t0.6\t1\t3 22\n",
      "79\t21\t0.6\t1\t3 18\n",
      "80\t31\t0.6\t1\t3 28\n",
      "81\t12\t0.6\t1\t0 12\n",
      "82\t32\t0.6\t1\t2 30\n",
      "83\t20\t0.6\t1\t2 18\n",
      "84\t15\t0.6\t1\t0 15\n",
      "85\t18\t0.6\t1\t0 18\n",
      "86\t13\t0.6\t1\t1 12\n",
      "87\t19\t0.6\t1\t0 19\n",
      "88\t19\t0.6\t1\t0 19\n",
      "89\t13\t0.6\t1\t0 13\n",
      "90\t19\t0.6\t1\t2 17\n",
      "91\t22\t0.6\t1\t0 22\n",
      "92\t21\t0.6\t1\t0 21\n",
      "93\t16\t0.6\t1\t0 16\n",
      "94\t32\t0.6\t1\t0 32\n",
      "95\t34\t0.6\t1\t0 34\n",
      "96\t19\t0.6\t1\t0 19\n",
      "97\t23\t0.6\t1\t1 22\n",
      "98\t16\t0.6\t1\t0 16\n",
      "99\t31\t0.6\t1\t0 31\n",
      "100\t24\t0.6\t1\t0 24\n",
      "101\t22\t0.6\t1\t0 22\n",
      "102\t36\t0.6\t1\t0 36\n",
      "103\t25\t0.6\t1\t0 25\n",
      "104\t37\t0.6\t1\t0 37\n",
      "105\t23\t0.6\t1\t1 22\n",
      "106\t34\t0.6\t1\t0 34\n",
      "107\t23\t0.6\t1\t1 22\n",
      "108\t12\t0.6\t1\t0 12\n",
      "109\t17\t0.6\t1\t1 16\n",
      "110\t16\t0.6\t1\t0 16\n",
      "111\t14\t0.6\t1\t1 13\n",
      "112\t17\t0.6\t1\t0 17\n",
      "113\t20\t0.6\t1\t0 20\n",
      "114\t19\t0.6\t1\t0 19\n",
      "115\t13\t0.6\t1\t0 13\n",
      "116\t12\t0.6\t1\t0 12\n",
      "117\t19\t0.6\t1\t2 17\n",
      "118\t18\t0.6\t1\t0 18\n",
      "119\t17\t0.6\t1\t0 17\n",
      "120\t11\t0.6\t1\t0 11\n",
      "121\t19\t0.6\t1\t0 19\n",
      "122\t13\t0.6\t1\t1 12\n",
      "123\t43\t0.6\t1\t0 43\n",
      "124\t19\t0.6\t1\t0 19\n",
      "125\t30\t0.6\t1\t0 30\n",
      "126\t12\t0.6\t1\t0 12\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC15N_CAGATC_L007_R2_001.fastq\n",
      "=============================================\n",
      "41714643 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files CC15N_CAGATC_L007_R1_001_trimmed.fq and CC15N_CAGATC_L007_R2_001_trimmed.fq\n",
      "file_1: CC15N_CAGATC_L007_R1_001_trimmed.fq, file_2: CC15N_CAGATC_L007_R2_001_trimmed.fq\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: CC15N_CAGATC_L007_R1_001_trimmed.fq and CC15N_CAGATC_L007_R2_001_trimmed.fq <<<<<\n",
      "Writing validated paired-end Read 1 reads to CC15N_CAGATC_L007_R1_001_val_1.fq\n",
      "Writing validated paired-end Read 2 reads to CC15N_CAGATC_L007_R2_001_val_2.fq\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of sequences analysed: 41714643\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 12692405 (30.43%)\n",
      "\n",
      "Deleting both intermediate output files CC15N_CAGATC_L007_R1_001_trimmed.fq and CC15N_CAGATC_L007_R2_001_trimmed.fq\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "CC17N_CGATGT_L008\n",
      "CC17N_CGATGT_L008_R1_001.fastq\n",
      "CC17N_CGATGT_L008_R2_001.fastq\n",
      "/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC17N_CGATGT_L008_R1_001.fastq\n",
      "/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/trimmNormals/\n",
      "Multicore support not enabled. Proceeding with single-core trimming.\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 3.4\n",
      "single-core operation.\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/trimmNormals/\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC17N_CGATGT_L008_R1_001.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t165086\tAGATCGGAAGAGC\t1000000\t16.51\n",
      "Nextera\t18\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t8\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 165086). Second best hit was Nextera (count: 18)\n",
      "\n",
      "Writing report to '/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/trimmNormals/CC17N_CGATGT_L008_R1_001.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC17N_CGATGT_L008_R1_001.fastq\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.6.4_dev\n",
      "Cutadapt version: 3.4\n",
      "Number of cores used for trimming: 1\n",
      "Quality Phred score cutoff: 35\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 3.4). Setting -j 1\n",
      "Writing final adapter and quality trimmed output to CC17N_CGATGT_L008_R1_001_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 35') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC17N_CGATGT_L008_R1_001.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "This is cutadapt 3.4 with Python 3.8.5\n",
      "Command line parameters: -j 1 -e 0.1 -q 35 -O 1 -a AGATCGGAAGAGC /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC17N_CGATGT_L008_R1_001.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 319.84 s (8 Âµs/read; 7.18 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              38,296,076\n",
      "Reads with adapters:                16,108,083 (42.1%)\n",
      "Reads written (passing filters):    38,296,076 (100.0%)\n",
      "\n",
      "Total basepairs processed: 4,825,305,576 bp\n",
      "Quality-trimmed:             993,932,392 bp (20.6%)\n",
      "Total written (filtered):  3,667,319,992 bp (76.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 16108083 times\n",
      "\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.8%\n",
      "  C: 29.1%\n",
      "  G: 26.3%\n",
      "  T: 16.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t6297864\t9574019.0\t0\t6297864\n",
      "2\t2064341\t2393504.8\t0\t2064341\n",
      "3\t692424\t598376.2\t0\t692424\n",
      "4\t348027\t149594.0\t0\t348027\n",
      "5\t227983\t37398.5\t0\t227983\n",
      "6\t287685\t9349.6\t0\t287685\n",
      "7\t291180\t2337.4\t0\t291180\n",
      "8\t211555\t584.4\t0\t211555\n",
      "9\t262533\t146.1\t0\t262187 346\n",
      "10\t184770\t36.5\t1\t182913 1857\n",
      "11\t170479\t9.1\t1\t168731 1748\n",
      "12\t216714\t2.3\t1\t214523 2191\n",
      "13\t192457\t0.6\t1\t190505 1952\n",
      "14\t213783\t0.6\t1\t211255 2528\n",
      "15\t188645\t0.6\t1\t186300 2345\n",
      "16\t206188\t0.6\t1\t203366 2822\n",
      "17\t243733\t0.6\t1\t240668 3065\n",
      "18\t152414\t0.6\t1\t150500 1914\n",
      "19\t188402\t0.6\t1\t186069 2333\n",
      "20\t148462\t0.6\t1\t146387 2075\n",
      "21\t242381\t0.6\t1\t238886 3495\n",
      "22\t166694\t0.6\t1\t164419 2275\n",
      "23\t148593\t0.6\t1\t146479 2114\n",
      "24\t190989\t0.6\t1\t188344 2645\n",
      "25\t139620\t0.6\t1\t137618 2002\n",
      "26\t134925\t0.6\t1\t133072 1853\n",
      "27\t152265\t0.6\t1\t149914 2351\n",
      "28\t170549\t0.6\t1\t168012 2537\n",
      "29\t149509\t0.6\t1\t147346 2163\n",
      "30\t116146\t0.6\t1\t114476 1670\n",
      "31\t117141\t0.6\t1\t115438 1703\n",
      "32\t119104\t0.6\t1\t117199 1905\n",
      "33\t122792\t0.6\t1\t120689 2103\n",
      "34\t127582\t0.6\t1\t125316 2266\n",
      "35\t141442\t0.6\t1\t138432 3010\n",
      "36\t152599\t0.6\t1\t150388 2211\n",
      "37\t94005\t0.6\t1\t92817 1188\n",
      "38\t37388\t0.6\t1\t36534 854\n",
      "39\t80932\t0.6\t1\t79657 1275\n",
      "40\t61176\t0.6\t1\t58055 3121\n",
      "41\t161016\t0.6\t1\t159632 1384\n",
      "42\t10412\t0.6\t1\t10178 234\n",
      "43\t6770\t0.6\t1\t6669 101\n",
      "44\t14584\t0.6\t1\t14397 187\n",
      "45\t29260\t0.6\t1\t28971 289\n",
      "46\t34365\t0.6\t1\t33978 387\n",
      "47\t44421\t0.6\t1\t43973 448\n",
      "48\t24623\t0.6\t1\t24317 306\n",
      "49\t38908\t0.6\t1\t38429 479\n",
      "50\t27439\t0.6\t1\t27065 374\n",
      "51\t32453\t0.6\t1\t32006 447\n",
      "52\t34011\t0.6\t1\t33367 644\n",
      "53\t29886\t0.6\t1\t29489 397\n",
      "54\t8435\t0.6\t1\t8176 259\n",
      "55\t14892\t0.6\t1\t14557 335\n",
      "56\t15664\t0.6\t1\t15345 319\n",
      "57\t14889\t0.6\t1\t14603 286\n",
      "58\t9429\t0.6\t1\t9235 194\n",
      "59\t11032\t0.6\t1\t10779 253\n",
      "60\t8773\t0.6\t1\t8260 513\n",
      "61\t11321\t0.6\t1\t10851 470\n",
      "62\t9474\t0.6\t1\t9249 225\n",
      "63\t7555\t0.6\t1\t7023 532\n",
      "64\t6168\t0.6\t1\t5648 520\n",
      "65\t4921\t0.6\t1\t4536 385\n",
      "66\t5162\t0.6\t1\t4663 499\n",
      "67\t4882\t0.6\t1\t4333 549\n",
      "68\t4653\t0.6\t1\t3890 763\n",
      "69\t4273\t0.6\t1\t3357 916\n",
      "70\t3736\t0.6\t1\t2706 1030\n",
      "71\t3273\t0.6\t1\t2206 1067\n",
      "72\t3091\t0.6\t1\t1779 1312\n",
      "73\t8245\t0.6\t1\t1530 6715\n",
      "74\t3486\t0.6\t1\t2113 1373\n",
      "75\t544\t0.6\t1\t456 88\n",
      "76\t26\t0.6\t1\t12 14\n",
      "77\t8\t0.6\t1\t1 7\n",
      "78\t14\t0.6\t1\t1 13\n",
      "79\t6\t0.6\t1\t1 5\n",
      "80\t8\t0.6\t1\t1 7\n",
      "81\t8\t0.6\t1\t0 8\n",
      "82\t8\t0.6\t1\t0 8\n",
      "83\t11\t0.6\t1\t0 11\n",
      "84\t8\t0.6\t1\t0 8\n",
      "85\t15\t0.6\t1\t0 15\n",
      "86\t12\t0.6\t1\t0 12\n",
      "87\t4\t0.6\t1\t0 4\n",
      "88\t13\t0.6\t1\t0 13\n",
      "89\t7\t0.6\t1\t0 7\n",
      "90\t6\t0.6\t1\t0 6\n",
      "91\t7\t0.6\t1\t0 7\n",
      "92\t6\t0.6\t1\t0 6\n",
      "93\t8\t0.6\t1\t0 8\n",
      "94\t8\t0.6\t1\t0 8\n",
      "95\t13\t0.6\t1\t0 13\n",
      "96\t14\t0.6\t1\t0 14\n",
      "97\t10\t0.6\t1\t0 10\n",
      "98\t10\t0.6\t1\t0 10\n",
      "99\t11\t0.6\t1\t0 11\n",
      "100\t19\t0.6\t1\t0 19\n",
      "101\t5\t0.6\t1\t0 5\n",
      "102\t8\t0.6\t1\t0 8\n",
      "103\t15\t0.6\t1\t0 15\n",
      "104\t7\t0.6\t1\t0 7\n",
      "105\t4\t0.6\t1\t0 4\n",
      "106\t11\t0.6\t1\t0 11\n",
      "107\t16\t0.6\t1\t0 16\n",
      "108\t12\t0.6\t1\t0 12\n",
      "109\t13\t0.6\t1\t0 13\n",
      "110\t13\t0.6\t1\t0 13\n",
      "111\t20\t0.6\t1\t0 20\n",
      "112\t14\t0.6\t1\t0 14\n",
      "113\t21\t0.6\t1\t0 21\n",
      "114\t20\t0.6\t1\t0 20\n",
      "115\t16\t0.6\t1\t0 16\n",
      "116\t9\t0.6\t1\t0 9\n",
      "117\t12\t0.6\t1\t0 12\n",
      "118\t3\t0.6\t1\t0 3\n",
      "119\t3\t0.6\t1\t0 3\n",
      "120\t7\t0.6\t1\t0 7\n",
      "121\t6\t0.6\t1\t0 6\n",
      "122\t6\t0.6\t1\t0 6\n",
      "123\t19\t0.6\t1\t1 18\n",
      "124\t7\t0.6\t1\t0 7\n",
      "125\t11\t0.6\t1\t0 11\n",
      "126\t28\t0.6\t1\t0 28\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC17N_CGATGT_L008_R1_001.fastq\n",
      "=============================================\n",
      "38296076 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to '/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/trimmNormals/CC17N_CGATGT_L008_R2_001.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC17N_CGATGT_L008_R2_001.fastq\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.6.4_dev\n",
      "Cutadapt version: 3.4\n",
      "Number of cores used for trimming: 1\n",
      "Quality Phred score cutoff: 35\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 3.4). Setting -j -j 1\n",
      "Writing final adapter and quality trimmed output to CC17N_CGATGT_L008_R2_001_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 35') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC17N_CGATGT_L008_R2_001.fastq <<< \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "This is cutadapt 3.4 with Python 3.8.5\n",
      "Command line parameters: -j 1 -e 0.1 -q 35 -O 1 -a AGATCGGAAGAGC /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC17N_CGATGT_L008_R2_001.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 318.33 s (8 Âµs/read; 7.22 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              38,296,076\n",
      "Reads with adapters:                15,400,848 (40.2%)\n",
      "Reads written (passing filters):    38,296,076 (100.0%)\n",
      "\n",
      "Total basepairs processed: 4,825,305,576 bp\n",
      "Quality-trimmed:           1,114,608,929 bp (23.1%)\n",
      "Total written (filtered):  3,564,882,738 bp (73.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 15400848 times\n",
      "\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.5%\n",
      "  C: 31.2%\n",
      "  G: 26.4%\n",
      "  T: 14.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t6469309\t9574019.0\t0\t6469309\n",
      "2\t1748216\t2393504.8\t0\t1748216\n",
      "3\t757920\t598376.2\t0\t757920\n",
      "4\t319631\t149594.0\t0\t319631\n",
      "5\t243022\t37398.5\t0\t243022\n",
      "6\t290513\t9349.6\t0\t290513\n",
      "7\t284544\t2337.4\t0\t284544\n",
      "8\t218363\t584.4\t0\t218363\n",
      "9\t291122\t146.1\t0\t290493 629\n",
      "10\t192810\t36.5\t1\t191126 1684\n",
      "11\t162536\t9.1\t1\t160822 1714\n",
      "12\t276078\t2.3\t1\t274076 2002\n",
      "13\t159690\t0.6\t1\t157841 1849\n",
      "14\t406821\t0.6\t1\t404368 2453\n",
      "15\t75667\t0.6\t1\t74784 883\n",
      "16\t128842\t0.6\t1\t127548 1294\n",
      "17\t299466\t0.6\t1\t297781 1685\n",
      "18\t45523\t0.6\t1\t44792 731\n",
      "19\t162052\t0.6\t1\t160925 1127\n",
      "20\t96365\t0.6\t1\t95588 777\n",
      "21\t78451\t0.6\t1\t77811 640\n",
      "22\t160719\t0.6\t1\t159231 1488\n",
      "23\t183216\t0.6\t1\t181101 2115\n",
      "24\t217173\t0.6\t1\t214904 2269\n",
      "25\t78614\t0.6\t1\t77742 872\n",
      "26\t130770\t0.6\t1\t129317 1453\n",
      "27\t123178\t0.6\t1\t121683 1495\n",
      "28\t130013\t0.6\t1\t128825 1188\n",
      "29\t61263\t0.6\t1\t60380 883\n",
      "30\t176345\t0.6\t1\t174697 1648\n",
      "31\t41360\t0.6\t1\t40895 465\n",
      "32\t90622\t0.6\t1\t89704 918\n",
      "33\t79800\t0.6\t1\t78899 901\n",
      "34\t111107\t0.6\t1\t109963 1144\n",
      "35\t72932\t0.6\t1\t72089 843\n",
      "36\t63885\t0.6\t1\t63084 801\n",
      "37\t107320\t0.6\t1\t106199 1121\n",
      "38\t56574\t0.6\t1\t55881 693\n",
      "39\t73472\t0.6\t1\t72671 801\n",
      "40\t54338\t0.6\t1\t53623 715\n",
      "41\t75043\t0.6\t1\t74074 969\n",
      "42\t91002\t0.6\t1\t90100 902\n",
      "43\t24504\t0.6\t1\t24088 416\n",
      "44\t53317\t0.6\t1\t52352 965\n",
      "45\t68549\t0.6\t1\t67718 831\n",
      "46\t28954\t0.6\t1\t28570 384\n",
      "47\t23743\t0.6\t1\t23347 396\n",
      "48\t30298\t0.6\t1\t29962 336\n",
      "49\t24303\t0.6\t1\t23959 344\n",
      "50\t25888\t0.6\t1\t25278 610\n",
      "51\t41374\t0.6\t1\t40951 423\n",
      "52\t11181\t0.6\t1\t10983 198\n",
      "53\t11865\t0.6\t1\t11696 169\n",
      "54\t12490\t0.6\t1\t12273 217\n",
      "55\t17935\t0.6\t1\t17626 309\n",
      "56\t11382\t0.6\t1\t11139 243\n",
      "57\t14699\t0.6\t1\t14363 336\n",
      "58\t14017\t0.6\t1\t13655 362\n",
      "59\t10990\t0.6\t1\t10685 305\n",
      "60\t12205\t0.6\t1\t11873 332\n",
      "61\t11431\t0.6\t1\t11062 369\n",
      "62\t10676\t0.6\t1\t10027 649\n",
      "63\t9892\t0.6\t1\t9367 525\n",
      "64\t7950\t0.6\t1\t7293 657\n",
      "65\t7034\t0.6\t1\t6206 828\n",
      "66\t6463\t0.6\t1\t5262 1201\n",
      "67\t12840\t0.6\t1\t4647 8193\n",
      "68\t9776\t0.6\t1\t9489 287\n",
      "69\t241\t0.6\t1\t201 40\n",
      "70\t27\t0.6\t1\t14 13\n",
      "71\t23\t0.6\t1\t1 22\n",
      "72\t21\t0.6\t1\t4 17\n",
      "73\t12\t0.6\t1\t1 11\n",
      "74\t28\t0.6\t1\t3 25\n",
      "75\t27\t0.6\t1\t0 27\n",
      "76\t15\t0.6\t1\t0 15\n",
      "77\t22\t0.6\t1\t0 22\n",
      "78\t26\t0.6\t1\t1 25\n",
      "79\t26\t0.6\t1\t0 26\n",
      "80\t36\t0.6\t1\t1 35\n",
      "81\t16\t0.6\t1\t1 15\n",
      "82\t19\t0.6\t1\t2 17\n",
      "83\t14\t0.6\t1\t0 14\n",
      "84\t17\t0.6\t1\t0 17\n",
      "85\t14\t0.6\t1\t1 13\n",
      "86\t19\t0.6\t1\t0 19\n",
      "87\t19\t0.6\t1\t1 18\n",
      "88\t11\t0.6\t1\t0 11\n",
      "89\t16\t0.6\t1\t0 16\n",
      "90\t10\t0.6\t1\t0 10\n",
      "91\t12\t0.6\t1\t0 12\n",
      "92\t15\t0.6\t1\t1 14\n",
      "93\t20\t0.6\t1\t0 20\n",
      "94\t34\t0.6\t1\t0 34\n",
      "95\t25\t0.6\t1\t0 25\n",
      "96\t14\t0.6\t1\t0 14\n",
      "97\t21\t0.6\t1\t0 21\n",
      "98\t21\t0.6\t1\t0 21\n",
      "99\t23\t0.6\t1\t0 23\n",
      "100\t23\t0.6\t1\t0 23\n",
      "101\t13\t0.6\t1\t0 13\n",
      "102\t22\t0.6\t1\t0 22\n",
      "103\t15\t0.6\t1\t0 15\n",
      "104\t29\t0.6\t1\t0 29\n",
      "105\t11\t0.6\t1\t0 11\n",
      "106\t38\t0.6\t1\t1 37\n",
      "107\t17\t0.6\t1\t0 17\n",
      "108\t16\t0.6\t1\t0 16\n",
      "109\t17\t0.6\t1\t0 17\n",
      "110\t6\t0.6\t1\t0 6\n",
      "111\t21\t0.6\t1\t0 21\n",
      "112\t14\t0.6\t1\t0 14\n",
      "113\t15\t0.6\t1\t0 15\n",
      "114\t14\t0.6\t1\t0 14\n",
      "115\t13\t0.6\t1\t0 13\n",
      "116\t7\t0.6\t1\t0 7\n",
      "117\t22\t0.6\t1\t0 22\n",
      "118\t22\t0.6\t1\t0 22\n",
      "119\t15\t0.6\t1\t0 15\n",
      "120\t7\t0.6\t1\t1 6\n",
      "121\t13\t0.6\t1\t1 12\n",
      "122\t20\t0.6\t1\t0 20\n",
      "123\t30\t0.6\t1\t0 30\n",
      "124\t20\t0.6\t1\t0 20\n",
      "125\t36\t0.6\t1\t0 36\n",
      "126\t85\t0.6\t1\t1 84\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/CC17N_CGATGT_L008_R2_001.fastq\n",
      "=============================================\n",
      "38296076 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files CC17N_CGATGT_L008_R1_001_trimmed.fq and CC17N_CGATGT_L008_R2_001_trimmed.fq\n",
      "file_1: CC17N_CGATGT_L008_R1_001_trimmed.fq, file_2: CC17N_CGATGT_L008_R2_001_trimmed.fq\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: CC17N_CGATGT_L008_R1_001_trimmed.fq and CC17N_CGATGT_L008_R2_001_trimmed.fq <<<<<\n",
      "Writing validated paired-end Read 1 reads to CC17N_CGATGT_L008_R1_001_val_1.fq\n",
      "Writing validated paired-end Read 2 reads to CC17N_CGATGT_L008_R2_001_val_2.fq\n",
      "\n",
      "Total number of sequences analysed: 38296076\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 8930126 (23.32%)\n",
      "\n",
      "Deleting both intermediate output files CC17N_CGATGT_L008_R1_001_trimmed.fq and CC17N_CGATGT_L008_R2_001_trimmed.fq\n",
      "\n",
      "====================================================================================================\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#once you change the prj folder, run this cell\n",
    "#!bash /<path>/testrun_trimgalorePaired_notebook.sh\n",
    "\n",
    "!bash /home/l.gonzalez/Notebooks/testrun_trimgalorePaired_notebook.sh\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run the aligner (STAR)\n",
    "In this case is a small script to run STAR with paired-end files\n",
    "Be aware of the path to the directory to change for the new one"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Variables to know where are the binary, references DO NOT CHANGE them\n",
    "STAR_BINARY=\"/home/l.gonzalez/programs/STAR-2.7.5c/bin/Linux_x86_64_static/STAR\"\n",
    "GENE_REFERENCE=\"/DATA/share/pipelines/lgleon/References/Ensembl/STAR_Ref/Homo_sapiens.GRCh38.99.gtf\"\n",
    "STAR_INDEX=\"//DATA/share/pipelines/lgleon/References/Ensembl/newSTAR_idx\"\n",
    "USE_THREADS=16\n",
    "\n",
    "#those ones you need to change with each analysis (run), here and in the script before run it\n",
    "WORKING_DIR = \"/DATA/share/pipelines/lgleon/C5_RNAseq_FFPE/\"\n",
    "DIR_INPUT_SAMPLES = \"/DATA/share/pipelines/lgleon/C5_RNAseq_FFPE/fastq_files\"\n",
    "DIR_BAM_OUTPUT = \"/DATA/share/pipelines/lgleon/C5_RNAseq_FFPE/bam_test_notrimm/\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The idea is run directly the small bash script /home/l.gonzalez/Notebooks/run_newpairedSTAR.sh\n",
    "# OR  /home/l.gonzalez/Notebooks/run_STAR_singleORpaired.sh\n",
    "# what is in this scrript: DO NOT RUN IT, IT WILL NOT WORK LIKE THAT\n",
    "\n",
    "#script:\n",
    "\n",
    "#!bin/bash\n",
    "\n",
    "#############################################################################\n",
    "##############################################################################\n",
    "# SINGLE-END part\n",
    "##############################################################################\n",
    "##############################################################################\n",
    "\n",
    "\n",
    "#where are the data that we are analyzing, this part is hard code need to be change each time\n",
    "prj=/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/test-code\n",
    "\n",
    "#loop to interact over all the fastq files in the folder, take one file\n",
    "#check the nale-tail of the files and adapt if need it\n",
    "for i in ${prj}/*R1*.f*q.gz \n",
    "\n",
    "do \n",
    "    name=$(basename ${i} _R1_001.fastq.gz) \n",
    "\n",
    "\techo ${name}\n",
    "\techo ${name}_R1_001.fastq.gz\n",
    "#\techo ${name}_R2_001.fastq.gz\n",
    "\techo ${prj}/output/\n",
    "\n",
    "\n",
    "    /home/l.gonzalez/programs/STAR-2.7.5c/bin/Linux_x86_64_static/STAR \\\n",
    "    --runMode alignReads \\\n",
    "    --genomeDir /DATA/share/pipelines/lgleon/References/Ensembl/newSTAR_idx/ \\\n",
    "    --runThreadN 16 \\\n",
    "    --sjdbGTFfile /DATA/share/pipelines/lgleon/References/Ensembl/STAR_Ref/Homo_sapiens.GRCh38.99.gtf \\\n",
    "    --readFilesIn  ${name}_R1_001.fastq.gz \\\n",
    "    --outSAMtype BAM SortedByCoordinate \\\n",
    "    --outFileNamePrefix ${prj}/output/ \\\n",
    "    --outFilterMismatchNmax 10 \\\n",
    "    --outFilterIntronMotifs None \\\n",
    "    --chimSegmentMin 1 \\\n",
    "    --sjdbScore 2 \\\n",
    "    --quantMode GeneCounts \\\n",
    "    --readFilesCommand zcat\n",
    "\n",
    "\n",
    "done\n",
    "\n",
    "# you can convert this one in parired end, \n",
    "# simple removing the # from the R2 and adding the second fastq in the input"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### To modified the file here, just open the sh file and do it, easy peasy \n",
    "We are in /home/l.gonzalez/Notebooks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CC15N_CAGATC_L007\n",
      "CC15N_CAGATC_L007_R1_001_val_1.fq\n",
      "CC15N_CAGATC_L007_R2_001_val_2.fq\n",
      "/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/trimmNormals/CC15N_CAGATC_L007_R1_001_val_1.fq\n",
      "/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/trimmNormals/output/\n",
      "/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/trimmNormals/output/CC15N_CAGATC_L007\n",
      "Jun 23 08:46:26 ..... started STAR run\n",
      "Jun 23 08:46:26 ..... loading genome\n",
      "Jun 23 08:46:44 ..... processing annotations GTF\n",
      "Jun 23 08:46:57 ..... inserting junctions into the genome indices\n",
      "Jun 23 08:48:25 ..... started mapping\n",
      "Jun 23 08:51:58 ..... finished mapping\n",
      "Jun 23 08:52:01 ..... started sorting BAM\n",
      "Jun 23 08:53:16 ..... finished successfully\n",
      "CC17N_CGATGT_L008\n",
      "CC17N_CGATGT_L008_R1_001_val_1.fq\n",
      "CC17N_CGATGT_L008_R2_001_val_2.fq\n",
      "/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/trimmNormals/CC17N_CGATGT_L008_R1_001_val_1.fq\n",
      "/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/trimmNormals/output/\n",
      "/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/fastq_normals/trimmNormals/output/CC17N_CGATGT_L008\n",
      "Jun 23 08:53:20 ..... started STAR run\n",
      "Jun 23 08:53:20 ..... loading genome\n",
      "Jun 23 08:53:38 ..... processing annotations GTF\n",
      "Jun 23 08:53:51 ..... inserting junctions into the genome indices\n",
      "Jun 23 08:55:20 ..... started mapping\n",
      "Jun 23 08:58:50 ..... finished mapping\n",
      "Jun 23 08:58:53 ..... started sorting BAM\n",
      "Jun 23 08:59:45 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "## Alignment with STAR\n",
    "## Go to the folder where you wanto to save the results\n",
    "## Once you changed the proj folder, run this cell\n",
    "\n",
    "#!bash /home/l.gonzalez/Notebooks/run_newpairedSTAR.sh \n",
    "#!bash /home/l.gonzalez/Notebooks/run_STAR_singleORpaired.sh\n",
    "\n",
    "#testing code \n",
    "#!bash /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/test-code/test_newpairedSTAR.sh\n",
    "!bash /home/l.gonzalez/Notebooks/run_newpairedSTAR.sh\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run featureCounts\n",
    "Again it is just a line in CLI, so go where you have the bam files and run it from there"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning: Unknown annotation format: GTF/GFF. GTF format is used.\n",
      "\n",
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.1\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m18 BAM files  \u001b[0m \u001b[0m                                  ||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-002-A_S1Aligned.sortedByCoord.out\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-003-A_S2Aligned.sortedByCoord.out\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-004-A_S3Aligned.sortedByCoord.out\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-006_S4Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m  ||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-008_S5Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m  ||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-009_S6Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m  ||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-015_S7Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m  ||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-017_S8Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m  ||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-020_S9Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m  ||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-027_S10Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m ||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-036_S21Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m ||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-040_S22Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m ||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-041_S11Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m ||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-045_S23Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m ||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-046_S12Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m ||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-047_S13Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m ||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-048_S24Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m ||\n",
      "||                           \u001b[32mo\u001b[36m B16PON-C-049_S14Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mfcounts_hg38_genename.counts\u001b[0m \u001b[0m                    ||\n",
      "||                 Summary : \u001b[36mfcounts_hg38_genename.counts.summary\u001b[0m \u001b[0m            ||\n",
      "||              Annotation : \u001b[36mHomo_sapiens.GRCh38.99.gtf (GTF)\u001b[0m \u001b[0m                ||\n",
      "||      Dir for temp files : \u001b[36m/DATA/share/pipelines/lgleon/organoids_rawda\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m16\u001b[0m \u001b[0m                                              ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mcounted\u001b[0m \u001b[0m                                         ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.99.gtf \u001b[0m... \u001b[0m                       ||\n",
      "||    Features : \u001b[36m60676\u001b[0m \u001b[0m                                                       ||\n",
      "||    Meta-features : \u001b[36m60642\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m47\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-002-A_S1Aligned.sortedByCoord.out.bam... \u001b[0m        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m55625758\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m46537824 (83.7%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.09 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-003-A_S2Aligned.sortedByCoord.out.bam... \u001b[0m        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m53106538\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m43785580 (82.4%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.08 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-004-A_S3Aligned.sortedByCoord.out.bam... \u001b[0m        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m52261896\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m43044474 (82.4%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.08 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-006_S4Aligned.sortedByCoord.out.bam... \u001b[0m          ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m103282045\u001b[0m \u001b[0m                                           ||\n",
      "||    Successfully assigned alignments : \u001b[36m37208720 (36.0%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.15 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-008_S5Aligned.sortedByCoord.out.bam... \u001b[0m          ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m55006036\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m45639579 (83.0%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.09 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-009_S6Aligned.sortedByCoord.out.bam... \u001b[0m          ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m50074880\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m41761024 (83.4%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.08 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-015_S7Aligned.sortedByCoord.out.bam... \u001b[0m          ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m49072196\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m42051946 (85.7%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.08 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-017_S8Aligned.sortedByCoord.out.bam... \u001b[0m          ||\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m47922223\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m39965068 (83.4%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.08 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-020_S9Aligned.sortedByCoord.out.bam... \u001b[0m          ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m62655466\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m52619003 (84.0%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.10 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-027_S10Aligned.sortedByCoord.out.bam... \u001b[0m         ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m59420369\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m50257247 (84.6%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.10 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-036_S21Aligned.sortedByCoord.out.bam... \u001b[0m         ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m54347595\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m44925977 (82.7%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.09 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-040_S22Aligned.sortedByCoord.out.bam... \u001b[0m         ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m56344219\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m47700132 (84.7%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.10 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-041_S11Aligned.sortedByCoord.out.bam... \u001b[0m         ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m53251989\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m44880162 (84.3%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.09 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-045_S23Aligned.sortedByCoord.out.bam... \u001b[0m         ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m61192937\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m52360705 (85.6%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.11 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-046_S12Aligned.sortedByCoord.out.bam... \u001b[0m         ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m78773066\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m66491600 (84.4%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.12 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-047_S13Aligned.sortedByCoord.out.bam... \u001b[0m         ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m48246269\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m40694377 (84.3%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.08 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-048_S24Aligned.sortedByCoord.out.bam... \u001b[0m         ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m70816151\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m58708845 (82.9%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.14 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file B16PON-C-049_S14Aligned.sortedByCoord.out.bam... \u001b[0m         ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m65671411\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m54879882 (83.6%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.10 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Write the final count table. \u001b[0m                                              ||\n",
      "|| Write the read assignment summary. \u001b[0m                                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"/DATA/share/pipelines/l \u001b[0m ||\n",
      "|| gleon/organoids_rawdata/RNAseq/counts/fcounts_hg38_genename.counts.summar \u001b[0m ||\n",
      "|| y\" \u001b[0m                                                                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## To run featureCounts is also a line that we can call frem here\n",
    "## go to the folder where all the bam files are store after algnment \n",
    "## Be aware:\n",
    "Binary = \"/home/l.gonzalez/programs/subread-2.0.1-source/bin/featureCounts\"\n",
    "GFT_Ref = \"/DATA/share/pipelines/lgleon/References/gencoderef/hg19/gencode.v19.annotation.gtf\"\n",
    "\n",
    "## run\n",
    "!/home/l.gonzalez/programs/subread-2.0.1-source/bin/featureCounts \\\n",
    "-T 16 -t gene -g gene_name -O -s 0 -M -Q 10 -a \\\n",
    "/DATA/share/pipelines/lgleon/References/Ensembl/STAR_Ref/Homo_sapiens.GRCh38.99.gtf -F GTF/GFF \\\n",
    "-o /DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/counts/fcounts_hg38_genename.counts \\\n",
    "/DATA/share/pipelines/lgleon/organoids_rawdata/RNAseq/bam_star/*bam\n",
    "\n",
    "#!/home/l.gonzalez/programs/subread-2.0.1-source/bin/featureCounts \\\n",
    "#-T 16 -t exon -g exon_id -O -s 0 -M -Q 10 -a \\\n",
    "#DATA/share/pipelines/lgleon/References/Ensembl/STAR_Ref/Homo_sapiens.GRCh38.99.gtf -F GTF/GFF \\\n",
    "#-o /DATA/share/pipelines/lgleon/NGS_protocol/RNA_Seq/Processed_Data/counts/NGS_normal_exon.counts \\\n",
    "#/DATA/share/pipelines/lgleon/NGS_protocol/RNA_Seq/Processed_Data/complete/Bam_normal_adjcentColon/*bam\n",
    "\n",
    "#Feature type(-t), 3rd column GTF file\n",
    "#CDS\n",
    "#Selenocysteine\n",
    "#exon\n",
    "#five_prime_utr\n",
    "#gene\n",
    "#start_codon\n",
    "#stop_codon\n",
    "#three_prime_utr\n",
    "#transcript\n",
    "\n",
    "#Atrivute that can be use (-g):\n",
    "#An example of attributes included in your GTF annotation is:\n",
    "#'gene_id \"ENSG00000223972\"; \n",
    "#gene_version \"5\"; \n",
    "#transcript_id \"ENST00000456328\"; \n",
    "#transcript_version \"2\"; \n",
    "#exon_number \"1\"; \n",
    "#gene_name \"DDX11L1\"; \n",
    "#gene_source \"havana\"; \n",
    "#gene_biotype \"transcribed_unprocessed_pseudogene\"; \n",
    "#transcript_name \"DDX11L1-202\"; \n",
    "#transcript_source \"havana\"; \n",
    "#transcript_biotype \"processed_transcript\"; \n",
    "#exon_id \"ENSE00002234944\"; \n",
    "#exon_version \"1\"; \n",
    "#tag \"basic\"; \n",
    "#transcript_support_level \"1\";' \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### After the analysis we will check the QC of the steps that we ran\n",
    "To do that, please go to the cell below and change the path to the folders where the fastqc files are, also the bam/bai files and multiqc will create a html with all the files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!multiqc /DATA/share/pipelines/lgleon/test/testCode_steps_Nousefuldata/testdata/fastq/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#for fname in glob.glob(DIR_INPUT_SAMPLES + \"/*.f*q.gz\"):\n",
    "#    print(fname)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
